packages/wallet/test/e2e/SingleAddressWallet/cip30mapping.test.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Wallet as Cip30Wallet,
+  RequestAccess,
+  WalletApi,
+  WalletProperties,
+  createUiWallet,
+  injectWindow
+} from '@cardano-sdk/cip30';
+import { SingleAddressWallet } from '../../../src';
+import { assetProvider, keyAgentReady, stakePoolSearchProvider, timeSettingsProvider, walletProvider } from '../config';
+import { createWalletApiAndHandleMessages } from '../../../src/util';
+import { firstValueFrom } from 'rxjs';
+import { mocks } from 'mock-browser';
+
+export const properties: WalletProperties = { apiVersion: '0.1.0', icon: 'imageLink', name: 'testWallet' };
+
+export const requestAccess: RequestAccess = async () => true;
+
+describe('cip30 e2e', () => {
+  let window: ReturnType<typeof mocks.MockBrowser>;
+  let wallet: SingleAddressWallet;
+  let injectedWallet: Cip30Wallet;
+  const windowStub = { ...window, location: { hostname: 'test-dapp' } };
+
+  beforeAll(async () => {
+    window = mocks.MockBrowser.createWindow();
+    // CREATE A WALLET
+    wallet = new SingleAddressWallet(
+      { name: 'Test Wallet' },
+      {
+        assetProvider,
+        keyAgent: await keyAgentReady,
+        stakePoolSearchProvider,
+        timeSettingsProvider,
+        walletProvider
+      }
+    );
+
+    const api: WalletApi = createUiWallet();
+    injectedWallet = new Cip30Wallet(properties, api, requestAccess);
+    injectWindow(window, injectedWallet);
+    createWalletApiAndHandleMessages(wallet, { logger: console, networkId: 0 });
+  });
+
+  afterAll(() => wallet.shutdown());
+
+  it('should correctly pull wallet balance correctly from blockfrost before starting the test', async () => {
+    await firstValueFrom(wallet.balance.total$);
+    expect(wallet.balance.total$.value?.coins).toBeGreaterThanOrEqual(0n);
+  });
+
+  describe('API Methods', () => {
+    let api: WalletApi;
+    beforeAll(async () => {
+      api = await window.cardano[properties.name].enable(windowStub.location.hostname);
+    });
+
+    xtest('api.getNetworkId', async () => {
+      const cip30NetworkId = await api.getNetworkId();
+      expect(cip30NetworkId).toEqual(0);
+    });
+
+    xtest('api.getUtxos', async () => {
+      const cip30WindowUtxos = await api.getUtxos();
+      const singleAddressWalletUtxos = await firstValueFrom(wallet.utxo.available$);
+      expect(cip30WindowUtxos).toEqual(singleAddressWalletUtxos);
+    });
+
+    test.skip('api.getBalance', async () => {
+      // const cip30Balance = await api.getBalance();
+      // const walletBalance = wallet.balance.total$.value;
+      // expect(cip30Balance).toEqual(walletBalance);
```

Comment:
Since cip30 defines all types as cbor, tests should probably use `cardano-serialization-lib` and assert it can parse the returned bytes, e.g.
```typescript
import * as CSL from '@emurgo/cardano-serialization-lib-nodejs'; // or wasm, or load dynamically in beforeAll

const balanceCborBytes = Buffer.from(await api.getBalance(), 'hex');
expect(() => CSL.Value.from_bytes(balanceCborBytes)).not.toThrow(); // I think asserting that it does not throw is sufficient
```